### PR TITLE
Remove Settings parameter from FieldMapper base class

### PR DESCRIPTION
**Pull request by romseygeek**
This is currently used to set the `indexVersionCreated` parameter on FieldMapper.
However, this parameter is only actually used by two implementations, and clutters
the API considerably.  We should just remove it, and use it directly in the 
implementations that require it.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/mapper/LegacyGeoShapeFieldMapper.java [hunk 1]
```diff
@@ -420,12 +420,15 @@ public PrefixTreeStrategy resolvePrefixTreeStrategy(String strategyName) {
...
+    private final Version indexCreatedVersion;
+
     public LegacyGeoShapeFieldMapper(String simpleName, FieldType fieldType, MappedFieldType mappedFieldType,
                                      Explicit<Boolean> ignoreMalformed, Explicit<Boolean> coerce, Explicit<Orientation> orientation,
                                      Explicit<Boolean> ignoreZValue, Settings indexSettings,
                                      MultiFields multiFields, CopyTo copyTo) {
-        super(simpleName, fieldType, mappedFieldType, ignoreMalformed, coerce, ignoreZValue, orientation, indexSettings,
+        super(simpleName, fieldType, mappedFieldType, ignoreMalformed, coerce, ignoreZValue, orientation,
             multiFields, copyTo);
+        this.indexCreatedVersion = Version.indexCreated(indexSettings);
```
Comment (cbuescher) [inline server/src/main/java/org/elasticsearch/index/mapper/LegacyGeoShapeFieldMapper.java:431]: I thought I'd never see a point where we use the settings at this stage in the PR ;-)

**Review by cbuescher — APPROVED** [review]
Nice clean up, LGTM. Couldn't find the second use btw...

File: x-pack/plugin/vectors/src/main/java/org/elasticsearch/xpack/vectors/mapper/DenseVectorFieldMapper.java [hunk 2]
```diff
@@ -144,10 +144,13 @@ public Query termQuery(Object value, QueryShardContext context) {
...
     }
 
+    private final Version indexCreatedVersion;
+
     private DenseVectorFieldMapper(String simpleName, FieldType fieldType, MappedFieldType mappedFieldType,
                                    Settings indexSettings, MultiFields multiFields, CopyTo copyTo) {
-        super(simpleName, fieldType, mappedFieldType, indexSettings, multiFields, copyTo);
+        super(simpleName, fieldType, mappedFieldType, multiFields, copyTo);
         assert fieldType.indexOptions() == IndexOptions.NONE;
+        this.indexCreatedVersion = Version.indexCreated(indexSettings);
```
Comment (romseygeek) [inline x-pack/plugin/vectors/src/main/java/org/elasticsearch/xpack/vectors/mapper/DenseVectorFieldMapper.java:153]: Here's the second use!

**Comment by romseygeek** [comment]
@elasticmachine update branch

**Pull request merged**